dsp-role/handler/src/main/scala/dsp/role/handler/RoleHandler.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.role.handler
+
+import dsp.errors.NotFoundException
+import dsp.errors.RequestRejectedException
+import dsp.role.api.RoleRepo
+import dsp.role.domain.Role
+import dsp.role.domain.RoleUser
+import dsp.valueobjects.Id.RoleId
+import dsp.valueobjects.Permission
+import dsp.valueobjects.Role._
+import zio.ZIO
+import zio._
+
+/**
+ * The role handler.
+ *
+ * @param repo  the role repository
+ */
+final case class RoleHandler(repo: RoleRepo) {
+
+  /**
+   * Retrieves all roles (sorted by IRI).
+   */
+  def getRoles(): UIO[List[Role]] =
+    repo
+      .getRoles()
+      .map(_.sorted)
+      .tap(_ => ZIO.logInfo("Retrieved all roles."))
+
+  /**
+   * Retrieves the role by ID.
+   *
+   * @param id of the role to be retrieved
+   */
+  def getRoleById(id: RoleId): IO[NotFoundException, Role] =
+    for {
+      role <- repo
+                .getRoleById(id)
+                .mapError(_ => NotFoundException(s"Role with ID: $id not found"))
+                .tap(_ => ZIO.logInfo(s"Found the role by ID: $id"))
+    } yield role
+
+  /**
+   * Creates a new role.
+   *
+   * @param name the role's name
+   * @param description the role's description
+   * @param users the role's users
+   * @param permission the role's permission
+   */
+  def createRole(
+    name: LangString,
+    description: LangString,
+    users: List[RoleUser],
+    permission: Permission
+  ): IO[Throwable, RoleId] =
+    (for {
+      id         <- RoleId.make().toZIO
+      role       <- Role.make(id, name, description, users, permission).toZIO
+      storedRole <- repo.storeRole(role)
+    } yield storedRole).tap(id => ZIO.logInfo(s"Created role with ID: $id"))
+
+  /**
+   * Updates the name of the role.
+   *
+   *  @param id  the roles's ID
+   *  @param newValue  the new name value
+   */
+  def updateName(id: RoleId, newValue: LangString): IO[RequestRejectedException, RoleId] =
```

Comment:
I am not sure if I like this `newValue` pattern. I know that we have it also in the user slice. But if the value is a list, it looks weird. So, maybe we should discuss this.

Reply:
I'm not sure if I get that comment. Param is always explained by its type and we also have it described in docstring. If the type would change to `Seq[LangString]` the `newValue` name for the param is still valid :) At the end it's still the same name translated to languages? We had this conversation before, so we should stick in this cases to singular. Or change the way we deal with internationalization.